source/connection-string/tests/invalid-options.yml

```
@@ -0,0 +1,126 @@
+tests:
+    -
+        description: "Error on unrecognized option keys"
+        uri: "mongodb://example.com/?foo=bar"
+        valid: false
+        warning: false
+        hosts:
+            -
+                type: "hostname"
+                host: "example.com"
+                port: ~
+        auth: ~
+        options: ~
+    -
+        description: "Error on unsupported option values"
+        uri: "mongodb://example.com/?fsync=ifPossible"
+        valid: false
+        warning: false
+        hosts:
+            -
+                type: "hostname"
+                host: "example.com"
+                port: ~
+        auth: ~
+        options: ~
+    -
+        description: "Repeated option keys with the same value"
+        uri: "mongodb://example.com/?replicaSet=test&replicaSet=test"
+        valid: false
+        warning: false
+        hosts:
+            -
+                type: "hostname"
+                host: "example.com"
+                port: ~
+        auth: ~
+        options:
+            replicaset: "test"
+
+    -
+        description: "Repeated option keys with different values"
+        uri: "mongodb://example.com/?replicaSet=test1&replicaSet=test2"
+        valid: false
+        warning: false
+        hosts:
+            -
+                type: "hostname"
+                host: "example.com"
+                port: ~
+        auth: ~
+        options:
+            replicaset: "test"
+    -
+        description: "List option with one empty option specified"
+        uri: "mongodb://example.com/?readPreferenceTags=dc:ny,rack:1&readPreferenceTags=dc:ny&readPreferenceTags="
+        valid: false
+        warning: false
+        hosts:
+            -
+                type: "hostname"
+                host: "example.com"
+                port: ~
+        auth: ~
+        options:
+            replicaset: "test"
```

Should be `null` as no `replicaSet` option is specified in the URI.

```suggestion
        options: ~
```